### ISSUE TITLE
Inconsistency between labels

### ISSUE THREAD

**Issue by niranjanakella** [openai/gpt-2-output-dataset#53]
I noticed an inconsistency between labels in the dataset.py file
File: detector/dataset.py
Lines: 47 - 58

**Before:**
```
def __getitem__(self, index):
        if self.epoch_size is not None:
            label = self.random.randint(2)
            texts = [self.fake_texts, self.real_texts][label]
            text = texts[self.random.randint(len(texts))]
        else:
            if index < len(self.real_texts):
                text = self.real_texts[index]
                label = 1
            else:
                text = self.fake_texts[index - len(self.real_texts)]
                label = 0
```

**What I feel should be:**
```
def __getitem__(self, index):
        if self.epoch_size is not None:
            label = self.random.randint(2)
            texts = [self.fake_texts, self.real_texts][label]
            text = texts[self.random.randint(len(texts))]
            label = 1 - label
        else:
            if index < len(self.real_texts):
                text = self.real_texts[index]
                label = 0
            else:
                text = self.fake_texts[index - len(self.real_texts)]
                label = 1
```
Now self.real_texts consistently have the label 0, and self.fake_texts have the label 1, regardless of whether self.epoch_size is None or not.

**Comment by niranjanakella** [comment]
This can be corrected by re-assigning the loss and logits to the corresponding model output values. 

**Before**
```
loss, logits = model(texts, attention_mask=masks, labels=labels)
```

**After**
```
model_out = model(texts, attention_mask=masks, labels=labels)
loss, logits = model_out.loss, model_out.logits```
